### PR TITLE
Update view-related api doc

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number:  #25495

## What is the new behavior?
Update for currant editorial usage and API doc style, include glossary references and proper term usage.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by mhevery — APPROVED** [review]

**Comment by jbogarthyde** [comment]
Need help passing circleci and g3 (if needed)

**Comment by jbogarthyde** [comment]
Need help passing circleci and g3 (if needed)

File: packages/core/src/linker/template_ref.ts [hunk 1]
```diff
@@ -11,31 +11,41 @@ import {EmbeddedViewRef} from './view_ref';
 
 
 /**
- * Represents an Embedded Template that can be used to instantiate Embedded Views.
+ * Represents an embedded template that can be used to instantiate embedded views.
+ * To instantiate embedded views based on a template, use {@link ViewContainerRef#
+ * createEmbeddedView}.
```
Comment (trotyl) [inline packages/core/src/linker/template_ref.ts]: The link here is broken due to line break.

File: packages/core/src/linker/template_ref.ts [hunk 2]
```diff
@@ -11,31 +11,41 @@ import {EmbeddedViewRef} from './view_ref';
...
+ * 
+ * Access a `TemplateRef` instance by placing a directive (or directive prefixed with `*`)
+ * on an `<ng-template>` element. The `TemplateRef` for the embedded view
```
Comment (trotyl) [inline packages/core/src/linker/template_ref.ts]: `(or directive prefixed with *)` should be placed after `on an <ng-template> element`, which is an alternative for the entire approach.

File: packages/core/src/linker/view_container_ref.ts [hunk 3]
```diff
@@ -15,111 +15,128 @@ import {EmbeddedViewRef, ViewRef} from './view_ref';
...
 
 /**
- * Represents a container where one or more Views can be attached.
+ * Represents a container where one or more views can be attached to a component.
  *
- * The container can contain two kinds of Views. Host Views, created by instantiating a
- * {@link Component} via {@link #createComponent}, and Embedded Views, created by instantiating an
- * {@link TemplateRef Embedded Template} via {@link #createEmbeddedView}.
+ * Can contain _host_ views (created by instantiating a
+ * component with the `createComponent()` method), and _embedded views_
```
Comment (trotyl) [inline packages/core/src/linker/view_container_ref.ts]: `_host_ views` -> `_host views_`? Should align with `_embedded views_` in either form.

File: packages/core/src/linker/view_container_ref.ts [hunk 4]
```diff
@@ -15,111 +15,128 @@ import {EmbeddedViewRef, ViewRef} from './view_ref';
...
+ * (created by instantiating a `TemplateRef` with the `createEmbeddedView()` method.
```
Comment (trotyl) [inline packages/core/src/linker/view_container_ref.ts]: Missing `)` at the end.

File: packages/core/src/linker/view_container_ref.ts [hunk 5]
```diff
@@ -15,111 +15,128 @@ import {EmbeddedViewRef, ViewRef} from './view_ref';
...
+   * @param index The 0-based index of the new location.
+   * @returns The moved `ViewRef` instance.
    */
   abstract move(viewRef: ViewRef, currentIndex: number): ViewRef;
 
   /**
-   * Returns the index of the View, specified via {@link ViewRef}, within the current container or
-   * `-1` if this container doesn't contain the View.
+   * Returns the index of a view within the current container.
+   * @parem viewRef The view to query.
```
Comment (trotyl) [inline packages/core/src/linker/view_container_ref.ts]: `parem` -> `param`?

File: packages/core/src/linker/view_ref.ts [hunk 6]
```diff
@@ -9,33 +9,51 @@
...
- * removing nested Views via a `ViewContainerRef`. Each View can contain many View Containers.
- * <!-- /TODO -->
+ * @see `ViewContainerRef`
  *
  * @usageNotes
- * ### Example
  *
- * Given this template...
+ * The following template breaks down into two separate `TemplateRef` instances,
+ * an outer one and and an inner one.
```
Comment (trotyl) [inline packages/core/src/linker/view_ref.ts]: Duplicate `and`?

**Comment by jenniferfell** [comment]
LGTM. We just need a googler to run the g3 presubmit tests for us. Thanks!

**Comment by benlesh** [comment]
No g3 presubmit required for documentation-only changes.

**Pull request closed without merging**